File: prjs/prog/prblm/clrs/csharp/CLRS/Part7/Chap32/StringMatching.cs
namespace CLRS.Part7.Chap32;
public class StringMatching
{
  public static int RabinKarp(string needle, string haystack){
	   var searcher = new RabinKarpSearch(needle);
	   return searcher.Search(haystack);
  }

  public static int KnuthMorrisPratt(string needle, string haystack){
	  var searcher = new KMPSearch(needle);
        return searcher.Search(haystack);
   }
}

File: prjs/prog/prblm/clrs/cpp/clrs/part7/chap32/string_matching.cpp
#include "string_matching.h"
#include "rabin_karp.h"
#include<iostream>
using namespace std;






int 

part7::chap32::
StringMatching::RabinKarp(string needle,string haystack){
   RabinKarpSearch searcher(needle);
   return searcher.Search(haystack);
}


